lib/internal/modules/esm/assert.js

```diff
@@ -46,54 +43,44 @@ const supportedAssertionTypes = ArrayPrototypeFilter(
 
 
 /**
- * Test a module's import assertions.
+ * Test a module's import attributes.
  * @param {string} url The URL of the imported module, for error reporting.
  * @param {string} format One of Node's supported translators
- * @param {Record<string, string>} importAssertions Validations for the
+ * @param {Record<string, string>} importAttributes Validations for the
  *                                                  module import.
  * @returns {true}
  * @throws {TypeError} If the format and assertion type are incompatible.
  */
-function validateAssertions(url, format,
-                            importAssertions = { __proto__: null }) {
+function validateAttributes(url, format,
+                            importAttributes = { __proto__: null }) {
   const validType = formatTypeMap[format];
 
-  if (!alreadyWarned && ObjectKeys(importAssertions).length !== 0) {
-    alreadyWarned = true;
-    process.emitWarning(
-      'Import assertions are not a stable feature of the JavaScript language. ' +
-      'Avoid relying on their current behavior and syntax as those might change ' +
-      'in a future version of Node.js.',
-      'ExperimentalWarning',
-    );
-  }
```

Comment:
this warning should still be preserved, especially for anyone using `assert` (which will continue to work in v8), but probably also for those using `with`, until it's clear that the proposal won't need to change further.

Reply:
There’s no way to differentiate import attributes that were defined with `assert` or with `with`, V8 gives us the same info regardless which keyword was used. If we keep the warning, we would need to keep it for both keywords. However I don’t think we want to do that because there’s already an experimental warning when using import assertions.

Reply:
Oh, hmm, that’s unfortunate. I think the warning should be kept for both somehow, and i agree it shouldn’t appear more than once - I’d ofc defer to you for the best way to achieve that :-)

Reply:
Imho, drop the warning. No other runtime besides node warns on language features, these are just annoying and make me set `--no-warnings` which also suppresses more important warnings than this one.

Reply:
It's still stage 3, and may still change.

Reply:
Why did it progress so far when it's considered so unstable?

Reply:
Stage 3 is always unstable in that way.

Reply:
The only use case for import attributes in Node.js today is JSON modules, which already have their own experimental warning. With this PR, any other use of import attribute would throw (per the spec proposal); the warning was added here only so folks who use custom import assertions are aware of the change of semantics.

Reply:
Sounds like anyone who uses an attribute will either get a syntax error or a warning, so that seems correct to me!